src/contract/ContractFunctionResult.ts

```diff
@@ -1,24 +1,93 @@
 import { ContractLogInfo, contractLogInfoListToSdk } from "./ContractLogInfo";
 import { ContractFunctionResult as ProtoContractFunctionResult } from "../generated/ContractCallLocal_pb";
 import { ContractId, contractIdToSdk } from "./ContractId";
+import BigNumber from "bignumber.js";
+import { encodeHex } from "../crypto/util";
 
-export type ContractFunctionResult = {
-    contractId: ContractId;
-    contractCallResult: Uint8Array | string;
-    errorMessage: string;
-    bloom: Uint8Array;
-    gasUsed: number;
-    logInfoList: ContractLogInfo[];
+export class ContractFunctionResult {
+    public readonly contractId: ContractId;
+    public readonly contractCallResult: Uint8Array;
+    public readonly errorMessage: string;
+    public readonly bloom: Uint8Array;
+    public readonly gasUsed: number;
+    public readonly logInfoList: ContractLogInfo[];
+
+    // Constructor isn't part of the stable API
+    public constructor(
+        contractId: ContractId,
+        contractCallResult: Uint8Array,
+        errorMessage: string,
+        bloom: Uint8Array,
+        gasUsed: number,
+        logInfoList: ContractLogInfo[]
+    ) {
+        this.contractId = contractId;
+        this.contractCallResult = contractCallResult;
+        this.errorMessage = errorMessage;
+        this.bloom = bloom;
+        this.gasUsed = gasUsed;
+        this.logInfoList = logInfoList;
+    }
+
+    public getString(index: number): string {
+        return Buffer.from(this.getBytes(index)).toString("utf-8");
+    }
+
+    public getBytes(index: number): Uint8Array {
+        // Len should never be larger than Number.MAX
+        // index * 32 is the position of the lenth
+        // index * 33 onward to index * 33 + (len * 32) will be the elements of the array
```

Comment:
bit of a math error here, `index * 32 + 32 != index * 33` (say, `index=10`, `10*32+32 = 320+32 = 352`, `10*33 = 330`) I believe you were looking for `(index + 1) * 32` or `index * 32 + 32`